addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -186,18 +186,22 @@ var MediaWidget = Widget.extend({
 var ImageWidget = MediaWidget.extend({
     template: 'web_editor.dialog.image',
     events: {
-        'click .o_upload_image_button': '_onUploadButtonClick',
+        'click .o_upload_image_button, .o_upload_document_button': '_onUploadButtonClick',
```

Comment:
Could you not share the logic like before ? Maybe rename both classes to 'o_upload_media_button' and share some of the xml template. Same for the URL button.

Reply:
Ok. Little question: isn't it a problem about

> trigger: ".o_select_media_dialog .o_upload_image_button",

with website_sale_tour_shop.js ?

Reply:
Well, when you rename something, you have to rename it everywhere. Including tours. Or you keep the old name. Your choice :wink: